src/main/java/com/puppycrawl/tools/checkstyle/checks/OrderedPropertiesCheck.java

```diff
@@ -0,0 +1,222 @@
+
+    /**
+     * Construct the check with default values.
+     */
+    public OrderedPropertiesCheck() {
+        setFileExtensions("properties");
+    }
+
+    /**
+     * Processes the file and check order.
+     * @param file the file to be processed
+     * @param fileText the contents of the file.
+     * @throws CheckstyleException
+     * I like to keep the order - keySet is not possible
+     * @noinspection EnumerationCanBeIteration
+     */
+    @Override
+    protected void processFiltered(File file, FileText fileText) throws CheckstyleException {
+        final SequencedProperties properties = new SequencedProperties();
+        try (InputStream inputStream = Files.newInputStream(file.toPath())) {
+            properties.load(inputStream);
+        }
+        catch (IOException ex) {
+            throw new CheckstyleException(UNABLE_TO_OPEN_FILE + file.getPath(), ex);
+        }
+
+        String previousProp = "";
+        int startLineNo = 0;
+
+        final Enumeration<Object> keys = properties.keys();
+
+        while (keys.hasMoreElements()) {
+
+            final String propKey = (String) keys.nextElement();
+
+            if (String.CASE_INSENSITIVE_ORDER.compare(previousProp, propKey) > 0) {
+
+                final int lineNo = getLineNumber(startLineNo, fileText, previousProp, propKey);
+                log(lineNo + 1, MSG_KEY, propKey, previousProp);
+                // start searching at position of the last reported validation
+                startLineNo = lineNo;
+            }
+
+            previousProp = propKey;
+        }
+    }
+
+    /**
+     * Method returns the line number where the key is detected.
+     * To assure that we get the correct line it starts at the point
+     * if the last occurrence.
+     * Also the previousProp should be in file before propKey.
+     *
+     * @param startLineNo start searching at line
+     * @param fileText {@link FileText} object contains the lines to process
+     * @param previousProp key name found last iteration, works only if valid
+     * @param propKey key name to look for
+     * @return line number of first occurrence. If no key found in properties file, 0 is returned
+     */
+    private static int getLineNumber(int startLineNo, FileText fileText,
+                                     String previousProp, String propKey) {
+        final Pattern prevKeyPattern = getKeyPattern(previousProp);
+        final Pattern keyPattern = getKeyPattern(propKey);
+
+        boolean prevKeyFound = false;
+        int indexKey = 0;
+
+        for (int index = startLineNo; index < fileText.size(); index++) {
+            final String line = fileText.get(index);
+            if (!prevKeyFound && prevKeyPattern.matcher(line).matches()) {
+
+                // find previous key, lock ahead for key
+                prevKeyFound = true;
+            }
+
+            if (prevKeyFound && keyPattern.matcher(line).matches()) {
```

Comment:
if `prevKeyFound` is set and `keyPattern.matcher` fails, shouldn't `prevKeyFound` also be reset?

Reply:
If it is reset, you would have to recheck `prevKeyPattern` since you could have something like:
````
b=1
b=2
a=0
````

Reply:
I added a test for this. prevKeyFound does not need to reset, because of the break.

Reply:
Done.